### PR TITLE
Alter features code to return similar ndarrays for cell, nucleus, cytoplasm

### PR DESCRIPTION
**Pull request by alanocallaghan**
Previously, if a cell had no cytoplasm (i.e., if nucleus mask == cell mask) then it would be dropped from the corresponding matrix of cytoplasmic features. This aims to ensure that the 3 matrices returned at the end are of the same dimensionality, so that it is easier to match features across cells.

I've done my best to ensure this works, but I'm not super confident in my torch-ing. I can add unit tests if you like

### CODE REVIEW

**Comment by ThibautGoldsborough** [comment]
Not sure about this line. If you try to onehot the LuCa1 image without doing it sparsely, you are going to run into big trouble.

```python
sparse_onehot = torch.nn.functional.one_hot(label.long(), num_classes=num_classes).flatten(0, 1).transpose(0, 1).to_sparse_coo().float()
```

**Comment by alanocallaghan** [comment]
Fair point. Can't say I really understand your sparse onehot implementation, but I added a `remap` argument, which allows it to be used when the label numbers need to be retained (e.g., for matching labels)

**Comment by alanocallaghan** [comment]
Actually a max arg makes more sense, since otherwise you can still have implicit dropping of labels if the missing labels happen to have high values.
